src/app/rundown-view/components/follow-playhead-timeline/follow-playhead-timeline.component.ts

```diff
@@ -77,6 +86,19 @@ export class FollowPlayheadTimelineComponent implements OnChanges {
     }
   }
 
+  private checkForSpanningElements(parts: readonly Part[]): Part[] {
```

Comment:
There is a mismatch between the name of the function and the return type. 

`checkFor` indicates more that it should return a boolean confirming if it has a spanning element or not. Not that it should return an array of Parts.

`elements` is quite vague. What elements are it checking for? Rundowns? Segments? Parts? Since the return type is `Part[]` it indicates the elements in question might also be Parts.

The name doesn't really indicate what happens if the check is true. You modify Pieces in this function, but that is not indicated by the name at all.